test/e2e/helpers.go

```diff
@@ -2,65 +2,166 @@ package e2e
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
+	"net/http"
 	"net/url"
+	"strings"
 
+	"github.com/Jeffail/gabs"
 	"github.com/openshift-knative/serverless-operator/test"
 	v1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/intstr"
 	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/spoof"
 )
 
-func setupMetricsRoute(caCtx *test.Context, name string) (*v1.Route, error) {
-	metricsRoute := &v1.Route{
-		ObjectMeta: meta.ObjectMeta{
-			Name:      "metrics-" + name,
-			Namespace: "openshift-serverless",
-		},
-		Spec: v1.RouteSpec{
-			Port: &v1.RoutePort{
-				TargetPort: intstr.FromString("8383"),
-			},
-			Path: "/metrics",
-			To: v1.RouteTargetReference{
-				Kind: "Service",
-				Name: "knative-openshift-metrics2",
-			},
-		},
-	}
-	r, err := caCtx.Clients.Route.Routes("openshift-serverless").Create(context.Background(), metricsRoute, meta.CreateOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("error creating service monitor route: %v", err)
+const servingNamespace = "knative-serving"
+
+type PrometheusInfo struct {
+	options   []interface{}
+	queryPath string
+	token     string
+}
+
+func NewPrometheusInfo(caCtx *test.Context) *PrometheusInfo {
```

Comment:
Similar to below, I'd rather pass the K8s client in here and return an error to make the code flow a bit clearer.